### PR TITLE
[Feature] (WIP) Updates the web checkout UI to be more inline with iOS standards

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

**Comment by sleroux** [comment]
Should be good for a review. Just waiting on getting a lock icon to use in place of the red image.

File: Assets/Plugins/iOS/Shopify/Checkout/Checkout.TitleView.swift [hunk 1]
```diff
@@ -24,47 +24,47 @@
...
-        layer.cornerRadius = cornerRadius
-        translatesAutoresizingMaskIntoConstraints = false
-        titleLabel?.font = UIFont.boldSystemFont(ofSize: 16)
+    override init(frame: CGRect) {
+        titleLabel = UILabel()
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.text = "checkout.shopify.com"
+        titleLabel.textColor = .darkGray
+        titleLabel.font = UIFont.systemFont(ofSize: 16)
+        titleLabel.sizeToFit()
```
Comment (dbart01) [inline Assets/Plugins/iOS/Shopify/Checkout/Checkout.TitleView.swift]: Is this necessary? Wouldn't an auto-layout pass with constraints assigned to the label automatically use it's intrinsic content size anyway?

File: Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift [hunk 2]
```diff
@@ -56,28 +57,73 @@ class WebViewController: UIViewController {
...
         webView.navigationDelegate = self
-        contentView.setActionButton(target: self, selector: #selector(didPressCancel))
-        
+        setupNavigationBar()
         loadCheckoutURL()
     }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        webView.addObserver(self, forKeyPath: #keyPath(WKWebView.estimatedProgress), options: .new, context: &kvoContext)
```
Comment (dbart01) [inline Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift]: I would put observer registration and unregistration in `init` and `dealloc` respectively. Having it in `viewWillAppear` and `viewWillDisappear` can cause weird behaviour when this view can still be visible after it "disappears".

File: Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift [hunk 3]
```diff
@@ -56,28 +57,73 @@ class WebViewController: UIViewController {
...
-        self.view = WebContentView(frame: view.frame)
+        self.view = WKWebView(frame: view.frame)
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         webView.navigationDelegate = self
-        contentView.setActionButton(target: self, selector: #selector(didPressCancel))
-        
+        setupNavigationBar()
```
Comment (dbart01) [inline Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift]: The contents of this method sets up views. It should probably be part of `override func loadView()` instead.

File: Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift [hunk 4]
```diff
@@ -87,21 +133,23 @@ extension Checkout.WebViewController {
...
     }
     
     public func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {}
     
     public func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
         if let url = webView.url, isThankYouPage(url) {
             delegate?.didLoadThankYouPage()
-            contentView.isFinished = true
+            actionButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(didTapDone))
+            navigationItem.rightBarButtonItem = actionButton
```
Comment (dbart01) [inline Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift]: Can this be encapsulated in a more intentional method? For example:
```swift
self.updateWithDoneButton()
```

**Pull request closed without merging**